crates/revm/src/evm_impl.rs

```diff
@@ -98,34 +98,32 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
     fn remove_l1_cost(
         is_deposit: bool,
         tx_caller: B160,
-        tx_l1_cost: Option<U256>,
+        l1_cost: U256,
         db: &mut DB,
         journal: &mut JournaledState,
     ) -> Result<(), EVMError<DB::Error>> {
         if is_deposit {
             return Ok(());
         }
-        if let Some(l1_cost) = tx_l1_cost {
-            let acc = journal
-                .load_account(tx_caller, db)
-                .map_err(EVMError::Database)?
-                .0;
-            if l1_cost.gt(&acc.info.balance) {
-                let x = l1_cost.as_limbs();
-                let u64_cost = if x[1] == 0 && x[2] == 0 && x[3] == 0 {
-                    x[0]
-                } else {
-                    u64::MAX
-                };
-                return Err(EVMError::Transaction(
-                    InvalidTransaction::LackOfFundForMaxFee {
-                        fee: u64_cost,
-                        balance: acc.info.balance,
-                    },
-                ));
-            }
-            acc.info.balance = acc.info.balance.saturating_sub(l1_cost);
+        let acc = journal
+            .load_account(tx_caller, db)
+            .map_err(EVMError::Database)?
+            .0;
+        if l1_cost.gt(&acc.info.balance) {
+            let x = l1_cost.as_limbs();
+            let u64_cost = if x[1] == 0 && x[2] == 0 && x[3] == 0 {
+                x[0]
+            } else {
+                u64::MAX
+            };
```

Comment:
wdyt about this instead?

```suggestion
            let u64_cost = if U256::from(u64::MAX).lt(&l1_cost) {
                u64::MAX
            } else {
                l1_cost.as_limbs()[0]
            };
```

Reply:
NOICE